### ISSUE TITLE
Grouped inputs do not announce legend to screen reader users

### ISSUE THREAD

**Issue by samsmith-workday** [Workday/canvas-kit#1060]
## 🐛 Bug Report

When a screen reader users moves focus to a grouped set of inputs the fieldset/legend relationship is not announced to a screen reader users. For example, if a group of checkboxes has a legend of "have you seen the new lego movie?" this will not be announced when focusing a checkbox with a label of "yes".

## To Reproduce

1. Navigate to story book
2. Open a grouped checkbox/radio button example
3. Turn on a screen reader (VO will work)
4. Navigate with arrow keys into the fieldset
5. Listen for no legend to be announced when entering the group or focusing a specific grouped input

## Expected Behavior

When moving focus into a fieldset using a screen reader users should hear both the legend text and label of the specific grouped input. 

## Actual Results

Currently the legend is not announced and there is no group announcement indication.

## Browser (if applicable)

Chrome

## Link to repl or repo (highly encouraged)

v4.7.0
![radio group screen shot](https://user-images.githubusercontent.com/64159840/117700228-9a6ea100-b17a-11eb-8972-74347f5c7129.jpg)

### Error Output

Paste the error output here

```bash

```

or

Include a screenshot of a visual bug

**Comment by sahlhoff** [comment]
Leaving a note here...This new implementation requires `<fieldset>` to be a flex container to direct the layout of `Top` and `Left` positions. In previous versions of browsers (firefox, chrome) `<fieldset>` would not accept flexbox layouts. This has been fixed in: https://bugzilla.mozilla.org/show_bug.cgi?id=1230207, https://bugs.chromium.org/p/chromium/issues/detail?id=375693
